inductiva/tests/test_output_dir_thread_safety.py

```diff
@@ -0,0 +1,25 @@
+"""Tests thread safety"""
+from concurrent.futures import ThreadPoolExecutor
+
+import inductiva
+
+
+def test_thread_safety():
+    """Tests that threads do not make definite changes to context."""
+    inductiva.set_output_dir("test_dir")
+    output_dir = inductiva.get_output_dir()
+
+    num_threads = 5
+    output_dirs = [f"output_dir_{i}" for i in range(num_threads)]
+
+    with ThreadPoolExecutor(max_workers=num_threads) as executor:
+        futures = [
+            executor.submit(inductiva.set_output_dir, output_dirs[i])
```

Comment:
we also need to check that the output dir is set to the correct value in each thread. Something along the lines:

```suggestion
            def test(out_dir):
                inductiva.set_output_dir(out_dir)
                assert inductiva.get_output_dir() == out_dir
                sleep some ms here
            executor.submit(test, output_dirs[i])
```

Also, try running with a bit more threads and adding a random sleep (tens to hundreds of  ms) in each test call to scramble things a bit and eventually discover potential concurrency issues.